core/vertx/src/main/java/io/thorntail/vertx/tracing/VertxEventBusInterceptor.java

```diff
@@ -0,0 +1,54 @@
+package io.thorntail.vertx.tracing;
+
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.Tracer;
+import io.opentracing.Tracer.SpanBuilder;
+import io.opentracing.propagation.Format;
+import io.opentracing.tag.Tags;
+import io.thorntail.TraceMode;
+import io.thorntail.vertx.VertxInitializer;
+import io.vertx.core.Handler;
+import io.vertx.core.eventbus.EventBus;
+import io.vertx.core.eventbus.SendContext;
+
+/**
+ * Intercepts {@link EventBus} send operations.
+ *
+ * @author Martin Kouba
+ * @see EventBus#addInterceptor(Handler)
+ */
+@SuppressWarnings("rawtypes")
+@Dependent
+public class VertxEventBusInterceptor implements Handler<SendContext> {
+
+    public static final String OP_NAME_SEND = "vertx-send";
+
+    @Inject
+    private Tracer tracer;
+
+    @Inject
+    private VertxInitializer vertxInitializer;
+
+    @Override
+    public void handle(SendContext sendContext) {
+        ActiveSpan parent = tracer.activeSpan();
```

Comment:
Again if this is running on event loop then getting span from thread locals would not work